### ISSUE TITLE
Type errors after upgrading to 4.3.2

### ISSUE THREAD

**Issue by danethomas** [barrelstrength/craft-sprout-seo#187]
### Description
Am seeing this on certain pages - not all though. 

This is with a multi locale site, so that may be impacting things. 

In admin when trying to save on SproutSEO > Website Identity > Save

```
TypeError: Return value of barrelstrength\sproutseo\models\Globals::getMeta() must be of the type array, object returned in /usr/share/nginx/html/craft/vendor/barrelstrength/sprout-seo/src/models/Globals.php:156
Stack trace:
#0 /usr/share/nginx/html/craft/vendor/barrelstrength/sprout-seo/src/models/Globals.php(115): barrelstrength\sproutseo\models\Globals->getMeta()
#1 /usr/share/nginx/html/craft/vendor/barrelstrength/sprout-seo/src/services/GlobalMetadata.php(100): barrelstrength\sproutseo\models\Globals->getGlobalByKey('meta', 'json')
#2 /usr/share/nginx/html/craft/vendor/barrelstrength/sprout-seo/src/controllers/GlobalMetadataController.php(179): barrelstrength\sproutseo\services\GlobalMetadata->saveGlobalMetadata(Array, Object(barrelstrength\sproutseo\models\Globals))
#3 [internal function]: barrelstrength\sproutseo\controllers\GlobalMetadataController->actionSaveGlobalMetadata()
#4 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#5 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#6 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('save-global-met...', Array)
#7 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('save-global-met...', Array)
#8 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Application.php(290): yii\base\Module->runAction('sprout-seo/glob...', Array)
#9 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Application.php(558): craft\web\Application->runAction('sprout-seo/glob...', Array)
#10 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Application.php(269): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#11 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#12 /usr/share/nginx/html/craft/web/index.php(21): yii\base\Application->run()
#13 {main}
```

On the front end when viewing a page:
```
TypeError: Return value of barrelstrength\sproutseo\helpers\OptimizeHelper::prepareAppendedTitleValue() must be of the type string, null returned in /usr/share/nginx/html/craft/vendor/barrelstrength/sprout-seo/src/helpers/OptimizeHelper.php:507
Stack trace:
#0 /usr/share/nginx/html/craft/vendor/barrelstrength/sprout-seo/src/services/Optimize.php(329): barrelstrength\sproutseo\helpers\OptimizeHelper::prepareAppendedTitleValue(Object(barrelstrength\sproutseo\models\Metadata), Object(barrelstrength\sproutseo\models\Globals))
#1 /usr/share/nginx/html/craft/vendor/barrelstrength/sprout-seo/src/services/Optimize.php(180): barrelstrength\sproutseo\services\Optimize->getPrioritizedMetadataModel(Object(craft\elements\Entry), Object(craft\models\Site))
#2 /usr/share/nginx/html/craft/vendor/barrelstrength/sprout-seo/src/services/Optimize.php(108): barrelstrength\sproutseo\services\Optimize->getMetadata(Object(craft\elements\Entry), Object(craft\models\Site), true, Array)
#3 /usr/share/nginx/html/craft/storage/runtime/compiled_templates/cc/cc6ff00417bba0473a73c4d8b2e006ba4bc3dbb6fc347088c72b4a4a35b6e147.php(45): barrelstrength\sproutseo\services\Optimize->getMetadataViaContext(Array)
#4 /usr/share/nginx/html/craft/vendor/twig/twig/src/Template.php(407): __TwigTemplate_8af0a2e710f2c302b83acfbe928c74319af9af1caf62294b8ac2486c21467220->doDisplay(Array, Array)
#5 /usr/share/nginx/html/craft/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#6 /usr/share/nginx/html/craft/storage/runtime/compiled_templates/c3/c333ec2bc9cd1f9e4d322fad59ec346005e15e5198a58e35ed385b2e2077975f.php(112): Twig\Template->display(Array, Array)
#7 /usr/share/nginx/html/craft/vendor/twig/twig/src/Template.php(407): __TwigTemplate_188ffc5791da8e543d7cb6f61357e0399e5c6436c1290fd0baf6c62d027eb5b3->doDisplay(Array, Array)
#8 /usr/share/nginx/html/craft/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#9 /usr/share/nginx/html/craft/vendor/twig/twig/src/Template.php(392): Twig\Template->display(Array)
#10 /usr/share/nginx/html/craft/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#11 /usr/share/nginx/html/craft/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#12 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/View.php(391): Twig\Environment->render('pricing', Array)
#13 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/View.php(452): craft\web\View->renderTemplate('pricing', Array)
#14 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Controller.php(235): craft\web\View->renderPageTemplate('pricing', Array, 'site')
#15 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/controllers/TemplatesController.php(98): craft\web\Controller->renderTemplate('pricing', Array)
#16 [internal function]: craft\controllers\TemplatesController->actionRender('pricing', Array)
#17 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#18 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#19 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('render', Array)
#20 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('render', Array)
#21 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Application.php(290): yii\base\Module->runAction('templates/rende...', Array)
#22 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('templates/rende...', Array)
#23 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Application.php(275): yii\web\Application->handleRequest(Object(craft\web\Request))
#24 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#25 /usr/share/nginx/html/craft/web/au/index.php(21): yii\base\Application->run()
#26 {main}
```

### Steps to reproduce

1. Everything has been worked fine
2. Upgrade to CraftCMS 3.4.5 and SproutSEO 4.3.2
3. Everything looked to be working on but then discovered these errrors
4. Have tried to disable and re-enable plugin, and re-save entries.

### Additional info

- Craft version: 3.4.5
- PHP version: 7.3.5
- Database driver & version: MySQL 5.5.5
- Plugins & versions: 4.3.2

**Comment by danethomas** [comment]
Also tried to uninstall and re-install on a local env and when I click on Upgrade to Pro I get this error:

```
yii\base\InvalidArgumentException: The file or directory to be published does not exist: /usr/share/nginx/html/craft/vendor/barrelstrength/sprout-seo/src/web/assets/base/dist in /usr/share/nginx/html/craft/vendor/yiisoft/yii2/web/AssetManager.php:455
Stack trace:
#0 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/AssetManager.php(52): yii\web\AssetManager->publish('/usr/share/ngin...')
#1 /usr/share/nginx/html/craft/vendor/twig/twig/src/Extension/CoreExtension.php(1495): craft\web\AssetManager->getPublishedUrl('@sproutseo/web/...', true)
#2 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/helpers/Template.php(96): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\web\AssetManager), 'getPublishedUrl', Array, 'method', false, false)
#3 /usr/share/nginx/html/craft/storage/runtime/compiled_templates/54/54a978cddf956da8583df4b3e06c154d6dba9af8bd3fb7bb389de22633143d39.php(48): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\web\AssetManager), 'getPublishedUrl', Array, 'method')
#4 /usr/share/nginx/html/craft/vendor/twig/twig/src/Template.php(407): __TwigTemplate_7d66d41f7552bcd01c4d6001c4e04136fdbeac3eba80c0b4aa861a436e73f602->doDisplay(Array, Array)
#5 /usr/share/nginx/html/craft/vendor/twig/twig/src/Template.php(380): Twig\Template->displayWithErrorHandling(Array, Array)
#6 /usr/share/nginx/html/craft/vendor/twig/twig/src/Template.php(392): Twig\Template->display(Array)
#7 /usr/share/nginx/html/craft/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#8 /usr/share/nginx/html/craft/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#9 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/View.php(391): Twig\Environment->render('sprout-seo/upgr...', Array)
#10 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/View.php(452): craft\web\View->renderTemplate('sprout-seo/upgr...', Array)
#11 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Controller.php(235): craft\web\View->renderPageTemplate('sprout-seo/upgr...', Array, 'cp')
#12 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/controllers/TemplatesController.php(98): craft\web\Controller->renderTemplate('sprout-seo/upgr...', Array)
#13 [internal function]: craft\controllers\TemplatesController->actionRender('sprout-seo/upgr...', Array)
#14 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#15 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#16 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('render', Array)
#17 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('render', Array)
#18 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Application.php(290): yii\base\Module->runAction('templates/rende...', Array)
#19 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('templates/rende...', Array)
#20 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/web/Application.php(275): yii\web\Application->handleRequest(Object(craft\web\Request))
#21 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#22 /usr/share/nginx/html/craft/web/index.php(21): yii\base\Application->run()
#23 {main}
```

**Comment by danethomas** [comment]
Looks like it's related to this commit https://github.com/barrelstrength/craft-sprout-seo/commit/7e19691b84da71d9e4884dfc1ed4bd54844500e9

**Comment by BenParizek** [comment]
@danethomas I've just released v4.3.3 which should resolve the `Globals::getMeta() must be of the type array` error and the `OptimizeHelper::prepareAppendedTitleValue() must be of the type string`.

Let me know if you see anything else wonky. I think the two errors, while similar, may have different root causes.

I was unable to recreate the  "Upgrade to Pro" error. I'm currently thinking that it may just have had something to do with one of the versions of the plugins being cached but if you run into it again let me know.

**Comment by danethomas** [comment]
Thanks @BenParizek - 4.3.3 is working as expected. 

Yeah I wouldn't worry about that upgrade pro issue - I suspect you're right. Once I ran into the issue I was trying out a couple of different things here so it's likely something with the environment.
